### PR TITLE
refactor: add ContextIsolator component

### PR DESCRIPTION
**Pull request by li-jia-nan**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor: add ContextIsolator component |
| 🇨🇳 Chinese | refactor: add ContextIsolator component |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
这个 size-limit 真是神奇

File: components/_util/ContextIsolator.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      {children}
+    </NoFormStyle>
+  </NoCompactStyle>
+);
+
+if (process.env.NODE_ENV !== 'production') {
+  ContextIsolator.displayName = 'ContextIsolator';
+}
+
+export default ContextIsolator;
```
Comment (afc163) [inline components/_util/ContextIsolator.tsx:18]: https://github.com/ant-design/ant-design/pull/49431#discussion_r1641660930

这个没支持上。

Comment (li-jia-nan) [reply components/_util/ContextIsolator.tsx:29]: done

**Comment by afc163** [comment]
看看 ci 问题。

**Review by afc163 — APPROVED** [review]

File: components/input-number/index.tsx [hunk 2]
```diff
@@ -137,8 +137,20 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
       suffix={suffixNode}
-      addonBefore={getInputAddon(addonBefore)}
-      addonAfter={getInputAddon(addonAfter)}
+      addonBefore={
+        addonBefore && (
+          <ContextIsolator isolateFormContext isolateSpaceContext>
+            {addonBefore}
+          </ContextIsolator>
+        )
+      }
```
Comment (afc163) [inline components/input-number/index.tsx:146]: 这里多了好多重复字节，getInputAddon 是不是还可以留着。

Comment (li-jia-nan) [reply components/input-number/index.tsx:146]: 我在本地跑了一下 dist，好像没区别，先合了

**Pull request merged**